app/models/conversation.rb

```diff
@@ -28,4 +28,16 @@ class Conversation < ApplicationRecord
   def recipient_user(current_user)
     current_user == user1 ? user2 : user1
   end
+
+  def opposite_user(user)
+    user1 == user ? user2 : user1
+  end
+
+  def unread_messages(current_user)
```

Comment:
it's not `current_user` a helper? So I assume you could not send it as params. Device provide it for you